tests/unit/insertTestResult.unitTest.ts

```diff
@@ -1223,6 +1223,78 @@ describe("insertTestResult", () => {
         });
     });
 
+    context("when inserting a non-adr HGV with null expiry Date and null certificateNumber", () => {
+        it("should not throw error", () => {
+            const testResult = {...testResultsPostMock[4]};
+            testResult.testTypes[0].testExpiryDate = null;
+            testResult.testTypes[0].certificateNumber = null;
+
+            MockTestResultsDAO = jest.fn().mockImplementation(() => {
+                return {
+                    createSingle: () => {
+                        return Promise.resolve(Array.of(testResultsPostMock[4]));
+                    },
+                    getTestNumber: () => {
+                        return Promise.resolve({ testNumber: "W01A00209", id: "W01", certLetter: "A", sequenceNumber: "002" });
+                    },
+                    getTestCodesAndClassificationFromTestTypes: () => {
+                        return Promise.resolve({
+                            linkedTestCode: "wde",
+                            defaultTestCode: "bde",
+                            testTypeClassification: "Annual With Certificate"
+                        });
+                    }
+                };
+            });
+
+            testResultsService = new TestResultsService(new MockTestResultsDAO());
+
+            expect.assertions(3);
+            return testResultsService.insertTestResult(testResult)
+                .then((insertedTestResult: any) => {
+                    expect(insertedTestResult[0].vehicleType).toEqual("hgv");
+                    expect(insertedTestResult[0].testResultId).toEqual("1113");
+                });
+        });
+    });
+
+    context("when inserting a cancelled adr HGV with null expiry Date and null certificateNumber", () => {
+        it("should not throw error", () => {
+            const testResult = {...testResultsPostMock[4]};
```

Comment:
suggest to use cloneDeep instead of spread operator.

Reply:
fixed